llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -1226,6 +1227,21 @@ void llvm::AIEPostRASchedStrategy::buildGraph(ScheduleDAGMI &DAG, AAResults *AA,
   static_cast<AIEScheduleDAGMI &>(DAG).recordDbgInstrs(Region);
 }
 
+bool AIEScheduleDAGMI::mayAlias(SUnit *SUa, SUnit *SUb, bool UseTBAA) {
+  BlockState &BS = getSchedImpl()->getInterBlock().getBlockState(getBB());
+  if (BS.FixPoint.Stage == SchedulingStage::Pipelining) {
+    int II = BS.FixPoint.II;
+    int IterA = SUa->NodeNum / II;
+    int IterB = SUb->NodeNum / II;
+    if (aliasAcrossVirtualUnrolls(SUa->getInstr(), SUb->getInstr(), IterA,
+                                  IterB) == AliasResult::NoAlias) {
+      return false;
+    }
+  }
+
+  return SUa->getInstr()->mayAlias(AAForDep, *SUb->getInstr(), UseTBAA);
```

Comment:
How about `return ScheduleDAGMI::mayAlias(SUa, SUb, UseTBAA)`

Reply:
> How about `return ScheduleDAGMI::mayAlias(SUa, SUb, UseTBAA)`

Yep, that would be better